src/Foreign/JavaScript/Server.hs

```diff
@@ -40,21 +40,33 @@ import Foreign.JavaScript.Types
 httpComm :: Config -> EventLoop -> IO ()
 httpComm Config{..} worker = do
     env <- getEnvironment
-    let portEnv = Safe.readMay =<< Prelude.lookup "PORT" env
-    let addrEnv = fmap BS.pack $ Prelude.lookup "ADDR" env
-    
-    let config = Snap.setPort      (maybe defaultPort id (jsPort `mplus` portEnv))
-               $ Snap.setBind      (maybe defaultAddr id (jsAddr `mplus` addrEnv))
-               $ Snap.setErrorLog  (Snap.ConfigIoLog jsLog)
-               $ Snap.setAccessLog (Snap.ConfigIoLog jsLog)
-               $ Snap.defaultConfig
+
+    let config = Snap.setErrorLog     (Snap.ConfigIoLog jsLog)
+               $ Snap.setAccessLog    (Snap.ConfigIoLog jsLog)
+               $ forceSSL env Snap.defaultConfig
 
     server <- Server <$> newMVar newFilepaths <*> newMVar newFilepaths <*> return jsLog
 
     Snap.httpServe config . route $
         routeResources server jsCustomHTML jsStatic
         ++ routeWebsockets (worker server)
 
+
+    where
+    forceSSL :: [(String, String)] -> Snap.Config m a -> Snap.Config m a
+    forceSSL env config =
+      if jsForceSSL
+        then Snap.setSSLBind      (maybe "0.0.0.0" id jsSSLBind)
+               $ Snap.setSSLPort      (maybe 443 id jsSSLPort)
+               $ Snap.setSSLCert      (maybe "cert.pem" id jsSSLCert)
+               $ Snap.setSSLKey       (maybe "key.pem" id jsSSLKey)
+               $ Snap.setSSLChainCert (maybe False id jsSSLChainCert) config
+        else do
+            let portEnv = Safe.readMay =<< Prelude.lookup "PORT" env
+                addrEnv = fmap BS.pack $ Prelude.lookup "ADDR" env
+             in Snap.setPort (maybe defaultPort id (jsPort `mplus` portEnv))
```

Comment:
```suggestion
             in Snap.setPort (fromMaybe defaultPort (jsPort `mplus` portEnv))
```

Could you use `fromMaybe` instead of `maybe … id`? (The original version is my fault, sorry for that.)